moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -667,6 +668,9 @@ void MoveItConfigData::outputFollowJointTrajectoryYAML(YAML::Emitter& emitter)
 // ******************************************************************************************
 bool MoveItConfigData::outputROSControllersYAML(const std::string& file_path)
 {
+  // Copy ros_control_config_ to a new vector to avoid modifying it
+  std::vector<ROSControlConfig> ros_controllers_config_output(ros_controllers_config_);
```

Comment:
Cand the copy of outputFollowJointTrajectoryYAML be moved into outputFollowJointTrajectoryYAML()? It seems it's not used elsewhere in the function. Also, can you add a comment as to why you are erasing things from the copy?

Reply:
addressed all comments.

Reply:
ros_controllers_config_output is not used in ``outputROSControllersYAML()``, only in ``outputFollowJointTrajectoryYAML()``. please move it there since its only actually used in that helper function

Reply:
`ros_controllers_config_output` is used in `outputROSControllersYAML()`  [here](https://github.com/ros-planning/moveit/pull/994/files#diff-b722193e66bd85583ac56b9d68186447R783) .